source/extensions/filters/common/rbac/matchers.cc

```diff
@@ -248,7 +258,11 @@ bool AuthenticatedMatcher::matches(const Network::Connection& connection,
 
 bool MetadataMatcher::matches(const Network::Connection&, const Envoy::Http::RequestHeaderMap&,
                               const StreamInfo::StreamInfo& info) const {
-  return matcher_.match(info.dynamicMetadata());
+  if (metadata_source_ == envoy::config::rbac::v3::MetadataSource::DYNAMIC) {
+    return matcher_.match(info.dynamicMetadata());
+  }
+
+  return matcher_.match(info.route()->metadata());
```

Comment:
How about the case where the route() is nullptr?

Reply:
That's a fair concern! I'll add a condition to handle that. Would matching on an empty metadata be appropriate in that case or should we fallback to matching on dynamic metadata? I personally think that we should match on empty metadata and maybe log a line in debug to convey that the route is missing.